src/main/java/org/repodriller/RepoVisitor.java

```diff
@@ -0,0 +1,168 @@
+ *   - {#link CommitVisitorIterator#endRepoTour} after you're done with this repo
+ *
+ * @author Mauricio Aniche
+ *
+ */
+public class RepoVisitor {
+
+	private Map<CommitVisitor, PersistenceMechanism> visitors;
+	private static final Logger log = Logger.getLogger(RepoVisitor.class);
+	private SCMRepository currentRepo; // One at a time.
+
+	public RepoVisitor() {
+		visitors = new HashMap<CommitVisitor, PersistenceMechanism>();
+		currentRepo = null;
+	}
+
+	/***
+	 * Methods for managing CommitVisitors.
+	 ***/
+
+	/**
+	 * Add this <CommitVisitor, PersistenceMechanism> pair to my collection.
+	 *
+	 * @param visitor
+	 * @param writer
+	 */
+	public void addVisitor(CommitVisitor visitor, PersistenceMechanism writer) {
+		visitors.put(visitor, writer);
+	}
+
+	/**
+	 * Remove this CommitVisitors from my collection, if present.
+	 *
+	 * @param visitor
+	 */
+	public void removeVisitor(CommitVisitor visitor) {
+		visitors.remove(visitor);
+	}
+
+	/**
+	 * Empty my collection of CommitVisitors.
+	 */
+	public void clearVisitors() {
+		visitors.clear();
+	}
+
+	/**
+	 * Print a description of all of the CommitVisitors I maintain.
+	 */
+	void printVisitors() {
+		for(CommitVisitor visitor : visitors.keySet()) {
+			log.info("- " + visitor.name() + " (" + visitor.getClass().getName() + ")");
+		}
+	}
+
+	/**
+	 * @return True if no CommitVisitors, else false.
+	 */
+	public boolean isEmpty() {
+		return visitors.isEmpty();
+	}
+
+	/***
+	 * Methods for visiting an SCMRepository.
+	 ***/
+
+	/**
+	 * Prepare for the next repo we visit.
+	 * Calls {@link CommitVisitor#initialize} on each CommitVisitor in my collection.
+	 *
+	 * @param repo	The repo we are visiting next
+	 */
+	void beginRepoVisit(SCMRepository repo) {
+		if (currentRepo != null)
+			throw new RepoDrillerException("Error, one repo at a time. We are already visiting a repo.");
+		currentRepo = repo;
+
+		for(Map.Entry<CommitVisitor, PersistenceMechanism> entry : visitors.entrySet()) {
+			CommitVisitor visitor = entry.getKey();
+			PersistenceMechanism writer = entry.getValue();
+
+			try {
+				log.info("-> Initializing visitor " + visitor.name());
+				visitor.initialize(currentRepo, writer);
+			} catch (Exception e) {
+				log.error("error in " + currentRepo.getPath() +
+						"when initializing " + visitor.name() + ", error=" + e.getMessage(), e);
+			}
+		}
+	}
+
+	/**
+	 * Visit this commit in the current repo.
+	 * Calls {@link CommitVisitor#process} on each CommitVisitor in my collection.
+	 *
+	 * @param commit
+	 */
+	void visitCommit(Commit commit) {
+		for(Map.Entry<CommitVisitor, PersistenceMechanism> entry : visitors.entrySet()) {
+			CommitVisitor visitor = entry.getKey();
+			PersistenceMechanism writer = entry.getValue();
+
+			try {
+				log.info("-> Processing " + commit.getHash() + " with " + visitor.name());
+				visitor.process(currentRepo, commit, writer);
+			} catch (CSVFileFormatException e) {
```

Comment:
not sure why this exception is here... This deserves an issue: #74

Reply:
Perhaps we should define what the behavior is if a CommitVisitor throws an exception.

Options:
1. We could remove the visitor from the RepoVisitor.
2. We can mask the exception.
3. We can throw.

Reply:
I believe the behavior is already correct there. If an exception happens, repodriller logs it, and moves forward. It could maybe notify the study or the visitors about the error.

But this particular CSVFileFormat, not sure what it does there. I think it halts the problem as soon as it notices that the output is not being saved. It makes sense to completely stop the execution. But we should make it more general, so that this behaviour is the same through all the persistence mechanisms! This should be in #74